run/generate_test.go

```diff
@@ -41,3 +44,75 @@ func TestAtomicGenerate(t *testing.T) {
 		t.Fatalf("Expected file to be unchanged, but was different.  Expected: %q, got: %q", original, b)
 	}
 }
+
+func TestCopyStaticFiles(t *testing.T) {
+	var originPaths []string
+	source := "static"
+	dest := "static_asset"
+	src := []struct {
+		path string
+		name string
+	}{
+		{"static/base", "base.md"},
+		{"static/base/level_one", "level_one.md"},
+		{"static/base/level_two", "level_two.md"},
+	}
+
+	for _, v := range src {
+		err := os.MkdirAll(v.path, 0777)
+		if err != nil {
+			t.Fatal(err)
+		}
+		o := filepath.Join(v.path, v.name)
+		err = ioutil.WriteFile(o, []byte(v.name), 0600)
+		if err != nil {
+			t.Fatal(err)
+		}
+		r, err := filepath.Rel(source, o)
+		if err != nil {
+			t.Fatal(err)
+		}
+		originPaths = append(originPaths, r)
+	}
+	defer func() {
+		err := os.RemoveAll("static")
+		if err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	err := copyStaticFiles(environ.Values{}, source, dest)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		err := os.RemoveAll(dest)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	// make sure the structure is preserved
+	var newPaths []string
+	filepath.Walk(dest, func(path string, info os.FileInfo, err error) error {
```

Comment:
since you have a well-known list of files you're looking for, you can probably use ioutil.ReadDir and the code will be easier to follow.

Reply:
We are checking the order of files in nested directories. ReadDir reads the content of one directories, I think walk here is a better fit rather than calling ReadDil a couple of times which we will also be walking the directory tree.